### PR TITLE
Employee schedules: endpoints and changes with reservations

### PR DESCRIPTION
**Pull request by GitaGliaudyte**
Added endpoints:
- Create employee schedule
- Delete employee schedule (marking isCancelled as true)
- Update employee schedule (Updates start or/and end time of selected schedule)
- Get employee schedule (Employee schedule by id, with availability)
- Get employee schedules by item and date (Self explanatory, lists every available user with availability)
- Get employee schedule (By employee id and date, with availability)
- Get all schedules by employee id (lists employee schedule without availability)

### CODE REVIEW

File: OmgvaPOS/ClientApp/dist/assets/index-Da-Q7oX2.js
Comment (AugustinasJarockis) [inline OmgvaPOS/ClientApp/dist/assets/index-Da-Q7oX2.js:1]: Shouldn't this file be ignored?

File: OmgvaPOS/ClientApp/package.json [hunk 1]
```diff
@@ -10,8 +10,6 @@
     "preview": "vite preview"
   },
   "dependencies": {
-    "@stripe/react-stripe-js": "^3.0.0",
-    "@stripe/stripe-js": "^5.2.0",
```
Comment (AugustinasJarockis) [inline OmgvaPOS/ClientApp/package.json:14]: Remove stripe?

File: OmgvaPOS/src/ScheduleManagement/Controller/ScheduleController.cs [hunk 2]
```diff
@@ -0,0 +1,92 @@
...
+        private readonly IScheduleService _scheduleService = scheduleService;
+
+        [HttpPost]
+        [Authorize(Roles = "Admin,Owner,Employee")]
+        [ProducesResponseType<EmployeeSchedule>(StatusCodes.Status201Created)]
+        [ProducesResponseType(StatusCodes.Status401Unauthorized)]
+        [ProducesResponseType(StatusCodes.Status500InternalServerError)]
+        public IActionResult CreateEmployeeSchedule([FromBody] CreateEmployeeScheduleRequest request)
+        {
+            var schedule = _scheduleService.CreateEmployeeSchedule(request);
```
Comment (AugustinasJarockis) [inline OmgvaPOS/src/ScheduleManagement/Controller/ScheduleController.cs:22]: Shouldn't there be checking for ability to manage business?

Comment (AugustinasJarockis) [reply OmgvaPOS/src/ScheduleManagement/Controller/ScheduleController.cs:22]: Same in other endpoints

File: OmgvaPOS/src/ScheduleManagement/Mappers/EmployeeScheduleMapper.cs [hunk 3]
```diff
@@ -0,0 +1,57 @@
+﻿using OmgvaPOS.ReservationManagement.Models;
+using OmgvaPOS.ScheduleManagement.DTOs;
+using OmgvaPOS.ScheduleManagement.Models;
+
+namespace OmgvaPOS.ScheduleManagement.Mappers
+{
+    public static class EmployeeScheduleMapper
+    {
+        public static Timeslot ToUnavailableTimeslot (Reservation reservation, TimeSpan duration)
```
Comment (AugustinasJarockis) [inline OmgvaPOS/src/ScheduleManagement/Mappers/EmployeeScheduleMapper.cs:9]: Not neccesarily in this PR, but in the future, could you use extension methods for mappers?

File: OmgvaPOS/src/ScheduleManagement/Validators/EmployeeScheduleValidator.cs [hunk 4]
```diff
@@ -0,0 +1,23 @@
...
+        public static void IsValidStartTime(TimeSpan reservationStart, TimeSpan timeToUpdate)
+        {
+            if (timeToUpdate > reservationStart)
+                throw new ValidationException("Start time cannot be later than the first reservation's start time.");
+        }
+        public static void IsValidEndTime(TimeSpan reservationEnd, TimeSpan timeToUpdate)
+        {
+            if (reservationEnd > timeToUpdate)
+                throw new ValidationException("End time cannot be earlier than the last reservation's end time.");
+        }
```
Comment (AugustinasJarockis) [inline OmgvaPOS/src/ScheduleManagement/Validators/EmployeeScheduleValidator.cs:21]: What is the difference between these three methods?

File: OmgvaPOS/ClientApp/package.json (hunk 1 above)
Comment (GitaGliaudyte) [reply OmgvaPOS/ClientApp/package.json:14]: Ahh, merge conflict issues..

File: OmgvaPOS/src/ScheduleManagement/Mappers/EmployeeScheduleMapper.cs (hunk 3 above)
Comment (GitaGliaudyte) [reply OmgvaPOS/src/ScheduleManagement/Mappers/EmployeeScheduleMapper.cs:9]: Noted!

File: OmgvaPOS/src/ScheduleManagement/Validators/EmployeeScheduleValidator.cs (hunk 4 above)
Comment (GitaGliaudyte) [reply OmgvaPOS/src/ScheduleManagement/Validators/EmployeeScheduleValidator.cs:21]: They are intented to validate start and end dates in different settings. As you can see, there are different exception messages here.

File: OmgvaPOS/src/ScheduleManagement/Controller/ScheduleController.cs (hunk 2 above)
Comment (GitaGliaudyte) [reply OmgvaPOS/src/ScheduleManagement/Controller/ScheduleController.cs:22]: I'm not sure of how I can aquire business id to compare it with the one from the token.

**Review by MrMnt — DISMISSED** [review]

**Review by MrMnt — APPROVED** [review]

**Review by AugustinasJarockis — APPROVED** [review]

**Pull request merged**